File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;

string solution(vector<int> food) {
	string leftStart;	
	string revLeftStart;

	for (int i = 1; i < food.size(); i++) {
		if (food[i]<2) continue;	//Check even		
		for (int j = 0; j < food[i] / 2; j++) {
			leftStart += to_string(i);
		}
	}
	revLeftStart = leftStart;
	reverse(revLeftStart.begin(), revLeftStart.end());
	leftStart += "0";
	leftStart += revLeftStart;
	return leftStart;
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.java
class Solution {
    public String solution(int[] food) {
        StringBuilder sb = new StringBuilder();
        
        for(int i=1;i<food.length;i++){
            int len = food[i];
            for(int j=0;j<len/2;j++){
                sb.append(i);
            }
        }
        StringBuilder sbAnswer = new StringBuilder(sb);
        sbAnswer.append(0);
        sbAnswer.append(sb.reverse());
        
        String answer = sbAnswer.toString();        
        return answer;
    }
}